src/core/thread/mle.cpp

```diff
@@ -1380,6 +1377,12 @@ void Mle::HandleParentRequestTimer(void)
                     {
                         // Restart keep-alive timer as it was disturbed by attachment procedure.
                         mParentRequestTimer.Start(0);
+
+                        // Return to rx-off-when-idle
+                        if ((mDeviceMode & ModeTlv::kModeRxOnWhenIdle) == 0)
+                        {
+                            mNetif.GetMeshForwarder().SetRxOnWhenIdle(false);
+                        }
```

Comment:
Yes, thanks for pointing that out.  I also cannot think of a reason why we need to address this case.